File: gcc/testsuite/gfortran.dg/pr57393-2.f90
! PR middle-end/57393
! { dg-do compile }
! { dg-options "-g -O2" }

SUBROUTINE pr57393 ( a1, a2, a3, a4, a5, a6, a7 )
  COMPLEX(kind=8), DIMENSION(:), INTENT(IN) :: a1
  INTEGER, DIMENSION(:), INTENT(IN) :: a2, a3, a5, a6
  COMPLEX(kind=8), DIMENSION(:), INTENT(INOUT) :: a4
  a4(a6(1)+1:a6(1)+a5(1))=a1(a3(1)+1:a3(1)+a2(1))
END SUBROUTINE pr57393

File: gcc/testsuite/gcc.dg/torture/pr57393-2.c
/* PR middle-end/57393 */
/* { dg-do compile } */

char a;

void
foo (int **p)
{
  int b;
  for (;;)
    {
      int c[1] = { 0 };
      unsigned *d = &c[0];
      for (b = 7; b; b--)
	**p &= --*d >= a;
    }
}
